### DISCUSSION TITLE
Where is this used?  gmail as way of sendmail?

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5273] [Q&A Core] [answered]
I found this in code:  (in admin/mailout.php)

```
		// Allow qmail as an option as well - works much as sendmail
		if ((strpos($_POST['sendmail'],'sendmail') !== FALSE) || (strpos($_POST['sendmail'],'qmail') !== FALSE))
		{
			$temp['sendmail'] = $tp->toDB($_POST['sendmail']);
		}
		else
		{
			$temp['sendmail'] = '';
		}

```
I can't find "sendmail" as the field of any form. It is commented out in mail_admin_class.php 

Yes, there are 3 mailers:
`$mailers = array('php' => 'php', 'smtp' => 'smtp', 'sendmail' => 'sendmail');
`
but it is field mailers.

I found:
`	$pref['sendmail'] - path to sendmail
`
Can be that code above commented too?  It isn't very clear.

**Comment by Jimmi08** [comment]
I noticed different logic for smtp options saving in preferences settings and in users/mail options.
In preferences saving keepalive and verp is working only if you set smtp options.  There is test if (!empty($_POST['smtp_options'])) for saving this. 
In Mails options it is saved for any smtp options.

**Reply by tgtje** [reply, marked as answer]
@Jimmi08 maybe this helps: also mentionded in mailout (general) : 
qmail can be used (if available) by selecting sendmail, and setting the sendmail path to that for qmail instead.
i.e. Unix systems;   Qmail would be more secure etc... it is like sendmail a transfer agent.

**Reply by Jimmi08** [reply]
Yes, I noticed this. But how you do this? There is setting only for smtp, not for sendmail.

**Reply by Deltik** [reply]
It appears that the "sendmail" pref represents the path to a [Sendmail](https://en.wikipedia.org/wiki/Sendmail) alternative called [qmail](https://en.wikipedia.org/wiki/Qmail).

Where this pref is used appears to be [here](https://github.com/e107inc/e107/blob/baa171cfb8de93947c6cfb7e6477cf558f94fce1/e107_handlers/mail.php#L336).

**Reply by Jimmi08** [reply]
Thanks, but I asked how to set the value of sendmail prefs in some form. Funny things is that I only now noticed that it is qmail and not gmail. Sorry,

**Reply by Deltik** [reply]
Sendmail and qmail are mail transfer agents that are configured at the server level by a sysadmin.  I imagine it would be a remote code execution vulnerability to allow them to be configured from a user-facing web form.
